src/main/java/browserview/BrowserComponent.java

```diff
@@ -530,6 +567,20 @@ public void switchToTab(String tabName) {
         }
         return Optional.empty();
     }
+
+    public void waitUntilDiscussionPageLoaded() {
+        for (int i = 0; i < WAIT_PAGE_LOAD_MAX_RETRY; i++) {
+            if (!isCurrentUrlDiscussion()) {
+                try {
+                    Thread.sleep(WAIT_PAGE_LOAD_SLEEP_TIME);
+                } catch (IllegalArgumentException | InterruptedException e) {
+                    logger.warn(e);
+                }
+            } else {
+                return;
+            }
+        }
```

Comment:
@acruis Here is my waiting. @dariusf suggested that should give limit to how long waiting so it won't get stuck.